tests/ui/test_ui_tools.py

```diff
@@ -2780,28 +2780,177 @@ def test_transform_content(self, mocker, raw_html, expected_content):
+                        "user": {
+                            "email": "iago@zulip.com",
+                            "full_name": "Iago",
+                            "id": 5,
+                        },
+                        "reaction_type": "unicode_emoji",
+                    },
+                    {
+                        "emoji_name": "zulip",
+                        "emoji_code": "zulip",
+                        "user": {
+                            "email": "AARON@zulip.com",
+                            "full_name": "aaron",
+                            "id": 1,
+                        },
+                        "reaction_type": "zulip_extra_emoji",
+                    },
+                    {
+                        "emoji_name": "zulip",
+                        "emoji_code": "zulip",
+                        "user": {
+                            "email": "shivam@zulip.com",
+                            "full_name": "Shivam",
+                            "id": 6,
+                        },
+                        "reaction_type": "unicode_emoji",
+                    },
+                ],
+                "expected_result": " :heart: Iago   :zulip: Shivam, You  ",
+                "reaction_attributes": [
+                    ("reaction", 14),
+                    (None, 1),
+                    ("reaction_mine", 21),
+                ],
+            },
+            {
+                "reactions": [
+                    {
+                        "emoji_name": "zulip",
+                        "emoji_code": "zulip",
+                        "user": {
+                            "email": "AARON@zulip.com",
+                            "full_name": "aaron",
+                            "id": 1,
+                        },
+                        "reaction_type": "zulip_extra_emoji",
+                    },
+                    {
+                        "emoji_name": "zulip",
+                        "emoji_code": "zulip",
+                        "user": {
+                            "email": "shivam@zulip.com",
+                            "full_name": "Shivam",
+                            "id": 6,
+                        },
+                        "reaction_type": "unicode_emoji",
+                    },
+                ],
+                "expected_result": " :zulip: Shivam, You  ",
+                "reaction_attributes": [
+                    ("reaction_mine", 21),
+                ],
+            },
         ],
     )
     def test_reactions_view(self, message_fixture, to_vary_in_each_message):
         self.model.user_id = 1
         varied_message = dict(message_fixture, **to_vary_in_each_message)
         msg_box = MessageBox(varied_message, self.model, None)
         reactions = to_vary_in_each_message["reactions"]
-
+        expected_result = to_vary_in_each_message["expected_result"]
+        reaction_attributes = to_vary_in_each_message["reaction_attributes"]
```

Comment:
This is what made me concerned about the test, as here we're changing the message we're testing with. That doesn't cause any issues here as it doesn't overwrite anything we expect to be there (and the Message is badly defined type-wise right now), but effectively we're generating a message with extra fields which the code could 'cheat' with, to pass the test ;)